apisix/plugins/kafka-logger.lua

```diff
@@ -205,7 +215,8 @@ function _M.log(conf, ctx)
     for host, port in pairs(conf.broker_list) do
         local broker = {
             host = host,
-            port = port
+            port = port,
+            sasl_config = conf.sasl_config or nil
```

Comment:
Would you please add test to cover it?
To make things simple, you can validate the behavior by injecting the producer.new and checking if sasl_config is passed through.

You can learn about the test framework with:
https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md